core/indexer/dataDispatcher.go

```diff
@@ -0,0 +1,105 @@
+package indexer
+
+import (
+	"context"
+	"time"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/core/indexer/workItems"
+)
+
+var log = logger.GetOrCreate("core/indexer")
+
+// Options structure holds the indexer's configuration options
+type Options struct {
+	IndexerCacheSize  int
+	UseKibana         bool
+	TxIndexingEnabled bool
+}
+
+const (
+	backOffTime = time.Second * 10
+	maxBackOff  = time.Minute * 5
+)
+
+type dataDispatcher struct {
+	backOffTime   time.Duration
+	chanWorkItems chan workItems.WorkItemHandler
+	cancelFunc    func()
+}
+
+// NewDataDispatcher creates a new dataDispatcher instance, capable of selecting the correct es that will
+//  handle saving different types
+func NewDataDispatcher(cacheSize int) (*dataDispatcher, error) {
+	dd := &dataDispatcher{
```

Comment:
should add a negative test for cacheSize

Reply:
tests will be added in a new branch

Reply:
no, should add a test here in productive code
```GO
if cacheSize < 0{
    return nil, ErrNegativeCacheSize
}
```

Reply:
done.